packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -1,22 +1,66 @@
 'use strict';
 
 const get = require('lodash/get');
+const semver = require('semver');
+
 const { parseSQSMessageBody, sendSQSMessage } = require('@cumulus/aws-client/SQS');
 const log = require('@cumulus/common/log');
-const { getMessageExecutionArn } = require('@cumulus/message/Executions');
+const {
+  getKnexClient,
+  Executions,
+} = require('@cumulus/db');
+const {
+  getMessageExecutionArn,
+  getMessageExecutionParentArn,
+  getMessageCumulusVersion,
+} = require('@cumulus/message/Executions');
 const Execution = require('../models/executions');
 const Granule = require('../models/granules');
 const Pdr = require('../models/pdrs');
 const { getCumulusMessageFromExecutionEvent } = require('../lib/cwSfExecutionEventUtils');
 
-const saveExecutionToDb = async (cumulusMessage) => {
+const isPostRDSDeploymentExecution = (cumulusMessage) => {
+  const minimumSupportedRDSVersion = process.env.RDS_DEPLOYMENT_CUMULUS_VERSION;
+  if (!minimumSupportedRDSVersion) {
+    throw new Error('RDS_DEPLOYMENT_CUMULUS_VERSION environment variable must be set');
+  }
+  const cumulusVersion = getMessageCumulusVersion(cumulusMessage);
+  return cumulusVersion
+    ? semver.gte(cumulusVersion, minimumSupportedRDSVersion)
+    : false;
+};
+
+const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
+  const executionIsPostDeployment = isPostRDSDeploymentExecution(cumulusMessage);
+  const parentArn = getMessageExecutionParentArn(cumulusMessage);
+  if (!executionIsPostDeployment || !parentArn) return executionIsPostDeployment;
+  return Executions.doesExecutionExist({
+    arn: parentArn,
+  }, knex);
+};
+
+const saveExecutions = async (cumulusMessage, knex) => {
   const executionModel = new Execution();
+  const executionArn = getMessageExecutionArn(cumulusMessage);
+
+  const isRDSWriteEnabled = await shouldWriteExecutionToRDS(cumulusMessage, knex);
+
+  if (!isRDSWriteEnabled) {
+    return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+  }
+
   try {
-    await executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+    return await knex.transaction(async (trx) => {
```

Comment:
I really went back and forth a lot on what Knex transaction style to use, but settled on this. I think we're generally trying to avoid callback style code, but I like that any error thrown in this callback rolls back the transaction and then automatically the Dynamo write code isn't reached.

There is also this style I considered, but having to explicitly call `trx.commit` and `trx.rollback` felt unnecessary. Please let me know if you disagree.

```
const trx = await knex.transaction();

const books = [
  {title: 'Canterbury Tales'},
  {title: 'Moby Dick'},
  {title: 'Hamlet'}
];

trx('catalogues')
  .insert({name: 'Old Books'}, 'id')
  .then(function(ids) {
    books.forEach((book) => book.catalogue_id = ids[0]);
    return trx('books').insert(books);
  })
  .then(trx.commit)
  .catch(trx.rollback);
```

Reply:
No, agreed, in this instance wrapping the transaction in an async callback block feels tidier.